docs/docs/how-tos/domain-registry.md

```diff
@@ -6,12 +6,81 @@ description: Setting a custom DNS Domain Registry
+
+2. You can outright buy a new domain with Cloudflare (this action isn’t particularly recommended).
+
+To generate a token [follow these steps](https://developers.cloudflare.com/api/tokens/create).
+
+- Under Profile, select the API Tokens menu and click on Create API Token.
+
+- On Edit zone DNS click on Use Template.![screenshot Cloudflare edit Zone DNS](../source/images/cloudflare_auth_1.png).
+
+- Configure Permissions such as the image below:![screenshot Cloudflare Permissions edit](../source/images/cloudflare_permissions_2.1.1.png)
+
+- On Account Resources set the configuration to include your desired account.![screenshot Cloudflare account resources](../source/images/cloudflare_account_resources_scr.png)
+
+- On Zone Resources set it to Include | Specific zone and your domain name.![screenshot Cloudflare zone resources](../source/images/cloudflare_zone_resources.png)
+
+- Click continue to summary.![screenshot Cloudflare summary](../source/images/cloudflare_summary.png)
+
+- Click on the Create Token button and set the token generated as an environment variable on your machine.
+ 
+
+Finally, set the environment variable such as:
+
+```bash
+ export CLOUDFLARE_TOKEN="cloudflaretokenvalue"
+```
+Also, add the flag `--dns-provider=cloudflare` to the [Qhub deploy command.](https://www.nebari.dev/how-tos/nebari-gcp#deploying-nebari)
+
+
+## Using other DNS providers
+
+Currently, QHub only supports CloudFlare for automatic DNS registration. If an alternate DNS provider is desired, change the `--dns-provider` flag from `cloudflare` to `none` on the qhub deploy command.
+
+Below are the links to detailed documentation on how to create and manage DNS records on a few providers:
+
+1. [Cloud DNS](https://cloud.google.com/dns/docs/tutorials/create-domain-tutorial) provider.
+2. [Amazon Route 53](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/domain-register.html) DNS provider.
+3. [Azure DNS](https://docs.microsoft.com/en-us/azure/dns/dns-getstarted-portal) provider.
+4. [Digital Ocean DNS](https://docs.digitalocean.com/products/networking/dns/quickstart/) provider.
+
+:::note
+Once your new DNS domain is created, it might take time for the records and related certs to fully propagate over the internet.
+
+The amount of time you might need to wait varies from each DNS provider. Validate such information in the related documentation for your chosen provider.
+:::
+
+## Automatically DNS provision
+
+Qhub has an extra flag for deployments that grants management and the creation of the DNS records for you automatically. For automatically DNS provision add `--dns-auto-provision` to your qhub deploy command:
+
+qhub deploy -c qhub-config `--dns-provider cloudflare --dns-auto-provision`
+This will set the DNS provider as Cloudflare and automatically handle the creation or updates for Qhub domain DNS records on Cloudflare.
```

Comment:
```suggestion
This will set the DNS provider as Cloudflare and automatically handle the creation or updates to the Qhub domain DNS records on Cloudflare.
```